### DISCUSSION TITLE
Support in myst for multiple directive arguments

### DISCUSSION THREAD

**Discussion by mmcky** [executablebooks/meta#155] [Ideas]
In Sphinx the `index` generating directive has the ability to use multiple arguments such as:

```rst
.. index::
   single: execution; context
   module: __main__
   module: sys
   triple: module; search; path

```

https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-index

I am not sure how to specify this use of multiple arguments in `myst`. As far as I am aware arguments need to be on the same line as the directive such as

````
```{index} single: execution; context
```
````

@chrisjsewell is there a way to specify multiple options at the moment?

---

I also note that in myst we use a similar syntax to specify options (albeit within the yaml headers only). 

````md
```{code-block}
---
class: <some class>
----
```
````

I don't see this as a big issue -- just something to be aware of. I think the `yaml` demarcators are a clear indicator that your specifying `options`.

**Comment by chrisjsewell** [comment]
have you tried just running?

````
```{index}
single: execution; context
module: __main__
module: sys
triple: module; search; path
```
````

**Comment by mmcky** [comment]
sorry should have mentioned that:

Using

````
```{index}
single: Python; IO
```
````

produces the following `warning`:

```bash
/Users/matthewmckay/repos-collab/lecture-python-programming.myst/source/python_essentials.md:274: WARNING: Directive 'index': 1 argument(s) required, 0 supplied

single: Python; IO
```

**Reply by chrisjsewell** [reply]
what about?

````
```{index} single: execution; context
module: __main__
module: sys
triple: module; search; path
```
````

I just seem to remember "fixing" multi-line argument strings, for use by sphinx-tabs

**Reply by mmcky** [reply]
Nice idea -- hadn't tried that continuation from the options line. Tried:

````
```{index} single: Python; Runtime
single: Julia
```
````

but get `No content permitted`:

```bash
/Users/matthewmckay/repos-collab/ebp-test-projectstructure/issues/index-directive/chapter3.md:3: WARNING: Directive 'index': No content permitted

single: Julia
```

**Comment by mmcky** [comment]
I was thinking we could have a line continuation marker, something like:

````
```{index} >
single: Python; Runtime
single: Julia
```
````

with content (if any in the future is needed) demarcated by a line break

````
```{index} >
single: Python; Runtime
single: Julia

<content>
```
````

and with `options`

````
```{index} >
single: Python; Runtime
single: Julia
---
name:
---
```
```` 

or 

````
```{index} >
single: Python; Runtime
single: Julia
:name: <targetname>
```
````

but not sure if that will cause issues on the parser side and the last example is a little confusing with `:` the only indicator of difference between argument and option.

**Reply by chrisjsewell** [reply]
the `>` could lead to issues, if you actually want to use `>` as the argument.
I think the logic would be something like: if the directive has a required argument, and the argument line is blank, try to take it from the next line(s) until the options start, or there is a blank line (I guess this is what docutils actually does)
